### ISSUE TITLE
Requests are returning error code 403

### ISSUE THREAD

**Issue by monecchi** [NiazMorshed2007/shadcn-address-autocomplete#1]
Hi there! Thanks for the awesome component!

I've been trying to implement it on my project but I keep getting "Error: HTTP error! status: 403" while requesting Google Places API...

I've checked for my Google API Console and indeed I have Google Places API (New) enabled for my project and `GOOGLE_PLACES_API_KEY` in place under .env, so technically that should work.

I mean, it is a bit hard to debug it because responses from Nextjs api routes `/app/api/address/autocomplete` and `/app/api/address/place` are ok with status 200, but no results from Google Places API though. 

Any ideas on how to circuvent that issue?

**Comment by monecchi** [comment]
I've managed to solve the issue by creating a new API KEY under Google API Console without any kind of restrictions.

It turns out I'm running it on **localhost** with **https** with along `next dev --experimental-https`, but even after whitelisting `https://localhost:3000` under Google API Console > Credentials settings, requests were being blocked somehow. I'm not quite sure if Google denies requests from localhost, but it worked!

**Comment by NiazMorshed2007** [comment]
Yep,, it's must to whitelist the domain where you'er requesting from.
This maybe can also occur because of experimental https of nextjs, but it worked for me to just whitelist `http://localhost:3000`

**Comment by kevinpruett-prime** [comment]
I too was receiving a 403. Turns out I needed to also pass through a `Referer` header in the request alongside `X-Goog-Api-Key`. I guess by default, Next.js was setting this `Referer` value to `about:client` which would not match my whitelisted domains
